src/lib/components/html-report-page.jsx

```diff
@@ -20,37 +20,55 @@ export type FlowUncoveredLocsProps = {
   uncovered_locs: Array<FlowUncoveredLoc>
 };
 
-/* eslint-disable react/no-unused-prop-types */
-export type FlowCoverageReportProps = {
-  reportType: 'summary' | 'sourcefile',
+export type FlowReportMetaProps = {
+  assets: {
+    css?: Array<string>,
+    js?: Array<string>
+  },
   coverageGeneratedAt: string,
+  htmlTemplateOptions: {
+    autoHeightSource?: boolean,
+  },
+};
+
+export type FlowCoverageSummaryReportProps = {
   assets: {
     css?: Array<string>,
     js?: Array<string>
   },
+  coverageGeneratedAt: string,
+  htmlTemplateOptions: {
+    autoHeightSource?: boolean,
+  },
```

Comment:
@ryan953 yeah, I see what you mean, it would be nicer if we could use the Intersection instead of the duplicate fields.

I played a bit with it and it seems that, while the type definition is valid, flow is not able to infer it correctly from the spread syntax used in the component (the `{...props}`).

Explicitly passing the "intersected" object properties into the react components seems to be enough to convince flow that all the expected properties has been passed:

```diff
diff --git a/src/lib/components/body-coverage-summary.jsx b/src/lib/components/body-coverage-summary.jsx
index 2f90f79..67e986f 100644
--- a/src/lib/components/body-coverage-summary.jsx
+++ b/src/lib/components/body-coverage-summary.jsx
@@ -71,7 +71,7 @@ export default function HTMLReportBodySummary(props: FlowCoverageSummaryReportPr
         </div>
         <div className="row">
           <h4 className="ui header">Summary</h4>
-          <FlowCoverageSummaryTable {...props}/>
+          <FlowCoverageSummaryTable {...props} coverageSummaryData={props.coverageSummaryData}/>
         </div>
         {
           meterBar
diff --git a/src/lib/components/html-report-page.jsx b/src/lib/components/html-report-page.jsx
index 500d8d8..3c22d2e 100644
--- a/src/lib/components/html-report-page.jsx
+++ b/src/lib/components/html-report-page.jsx
@@ -31,28 +31,11 @@ export type FlowReportMetaProps = {
   },
 };
 
-export type FlowCoverageSummaryReportProps = {
-  assets: {
-    css?: Array<string>,
-    js?: Array<string>
-  },
-  coverageGeneratedAt: string,
-  htmlTemplateOptions: {
-    autoHeightSource?: boolean,
-  },
+export type FlowCoverageSummaryReportProps = FlowReportMetaProps & {
   coverageSummaryData: FlowCoverageSummaryData,
 };
 
-export type FlowCoverageSourceFileReportProps = {
-  assets: {
-    css?: Array<string>,
-    js?: Array<string>
-  },
-  coverageGeneratedAt: string,
-  htmlTemplateOptions: {
-    autoHeightSource?: boolean,
-  },
-  coverageSummaryData: FlowCoverageSummaryData,
+export type FlowCoverageSourceFileReportProps = FlowCoverageSummaryReportProps & {
   coverageData: FlowCoverageJSONData,
   fileName: string,
   fileContent: string|Buffer,
@@ -62,13 +45,20 @@ export type FlowCoverageSourceFileReportProps = {
 export function HTMLReportSummaryPage(props: FlowCoverageSummaryReportProps) {
   return (<html>
     <HTMLReportHead {...props}/>
-    <HTMLReportBodySummary {...props}/>
+    <HTMLReportBodySummary {...props} coverageSummaryData={props.coverageSummaryData}/>
   </html>);
 }
 
 export function HTMLReportSourceFilePage(props: FlowCoverageSourceFileReportProps) {
   return (<html>
     <HTMLReportHead {...props}/>
-    <HTMLReportBodySourceFile {...props}/>
+    <HTMLReportBodySourceFile
+      {...props}
+      coverageSummaryData={props.coverageSummaryData}
+      coverageData={props.coverageData}
+      fileName={props.fileName}
+      fileContent={props.fileContent}
+      summaryRelLink={props.summaryRelLink}
+      />
   </html>);
 }
```